src/software/geom/shot.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "software/geom/point.h"
+
+class Shot
+{
+   public:
+    /**
+     * Creates a shot with the given target and angle
+     *
+     * @param point The target to shoot at
+     * @param angle The angle between the obstacles on either side of the shot vector
+     *
+     */
+    Shot(Point point, Angle angle);
+
+    /**
+     * The target of the shot
+     *
+     */
+    Point point;
+
+    /**
+     * The angle between the obstacles on either side of the shot vector
+     *
+     */
+    Angle angle;
```

Comment:
It seems that the class is being used as a struct since the code outside the class field can have direct access to member variables. I am not quite sure whether this implementation is the intention of this class or not. But if the intention is using a class rather than a struct, then `point` and `angle` should be under a `private` specifier to prevent accidental access and modification of those two member variables. Functions like `getPoint()` and `getAngle()` should be added under `public` specifier to allow the code outside the class to access the member variables. I would recommend looking at one of the 'Primitives' under `ai/primitive` as examples.

Reply:
Agreed, initially I thought why not use a struct? But there was some discussion on the issue ticket #771, Mathew and Gareth were leaning towards making this a class for now and it would be easier to add more functionality to it, if we wanted to, later on. 

As for why members are public, I was maintaining previous behavior which was that std::pair<Point, Angle> had both point and angle as basically public fields that were accessible and modifiable. 

I'm open to changing the fields to private instead and be unmodifiable. Maybe we can wait for the other reviewers?

Reply:
I did a quick scan through all the uses, and it seems like we never modify those values once they are created, so its pretty safe to make them private and provide accessors for them. If we wanted to add functionality later or change the internal representation of how we store/do stuff with point or angle, we can do it in the `point()` and `angle()` functions

Reply:
Sounds good, updated PR. Thanks guys